src/app/task.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import * as TaskJson from './task_sample_data.json';
+import { TaskData } from './taskdata';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskService {
+
+  taskList = TaskJson;
+
+  constructor() { }
+
+  getAllTasks() {
+    return this.taskList.tasks;
+  }
+
+  deleteTask(id: number) {
+    const index = this.taskList.tasks.findIndex(t => t.id === id);
```

Comment:
easier would be here for the time being to use this.taskList.tasks.filter, filter out by taskId and return it in 1 single line :)